subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CachedCustomTaskExecutionIntegrationTest.groovy

```diff
@@ -648,6 +649,36 @@ class CachedCustomTaskExecutionIntegrationTest extends AbstractIntegrationSpec i
         skippedTasks.sort() == [":consumer", ":producer"]
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/3043")
+    def "URL-quoted characters in file names are handled properly"() {
+        def weirdOutputPath = 'build/bad&dir/bad! <Dezső> %20.txt'
+        def expectedOutput = file(weirdOutputPath)
+        buildFile << """
+            task weirdOutput {
+                outputs.dir("build")
+                outputs.cacheIf { true }
+                doLast {
+                    mkdir file('$weirdOutputPath').parentFile
```

Comment:
Doesn't the directory now get created automatically? So we can just remove this line?

Reply:
`build` gets created automatically, `build/bad&dir` does not.